vault/eventbus/bus.go

```diff
@@ -127,7 +189,7 @@ func (bus *EventBus) Subscribe(_ context.Context, eventType logical.EventType) (
 
 	pipeline := eventlogger.Pipeline{
 		PipelineID: eventlogger.PipelineID(pipelineID),
-		EventType:  eventlogger.EventType(eventType),
+		EventType:  nsEventTypeJoin(ns, eventType),
```

Comment:
It feels like we have the wrong pipeline/node architecture if we're having to tack namespaces onto event type names, which are two pretty different concepts.

How about embedding the namespace in an async node's internal state, and then it can drop events on the floor when they don't match its own namespace? I think that should be flexible enough to support processing events from child namespaces in the future too, which I can imagine is a likely feature request. That also seems like it will extend to filtering on path too, which I think we'll want?

Reply:
Okay, I'll give this a shot. I agree that what I have now is not the right approach. :)